compose/service.py

```diff
@@ -857,3 +893,12 @@ def build_extra_hosts(extra_hosts_config):
         "extra_hosts_config \"%s\" must be either a list of strings or a string->string mapping," %
         extra_hosts_config
     )
+
+
+class NoSuchService(Exception):
```

Comment:
This one was defined in ``project.py`` so moved here to avoid circular dependency.